fabric-resource-loader-v0/src/main/java/net/fabricmc/fabric/impl/resource/loader/ModResourcePackUtil.java

```diff
@@ -82,6 +92,54 @@ public static InputStream openDefault(ModMetadata info, ResourceType type, Strin
 		}
 	}
 
+	// Basically a copy of the vanilla method, with an additional parameter for Text
+	@Nullable
+	public static ResourcePackProfile of(String name, Text text, boolean alwaysEnabled, Supplier<ResourcePack> packFactory, ResourcePackProfile.Factory profileFactory, ResourcePackProfile.InsertionPosition insertionPosition, ResourcePackSource packSource) {
+		try {
+			ResourcePack resourcePack = packFactory.get();
+
+			ResourcePackProfile var8;
+
+			label54: {
+				try {
+					PackResourceMetadata packResourceMetadata = resourcePack.parseMetadata(PackResourceMetadata.READER);
+
+					if (packResourceMetadata != null) {
+						var8 = profileFactory.create(name, text, alwaysEnabled, packFactory, packResourceMetadata, insertionPosition, packSource);
+						break label54;
+					}
+
+					LOGGER.warn("Couldn't find pack meta for pack {}", name);
+				} catch (Throwable var10) {
+					if (resourcePack != null) {
+						try {
+							resourcePack.close();
+						} catch (Throwable var9) {
+							var10.addSuppressed(var9);
+						}
+					}
+
+					throw var10;
+				}
+
+				if (resourcePack != null) {
+					resourcePack.close();
+				}
+
+				return null;
+			}
```

Comment:
Please use try-with-resources instead of this mess (this is just how Fernflower decompiles it)

Reply:
> Please use try-with-resources instead of this mess (this is just how Fernflower decompiles it)

resolved